openfold/utils/loss.py

```diff
@@ -1649,9 +1665,9 @@ def chain_center_of_mass_loss(
     all_atom_pred_pos = all_atom_pred_pos[..., ca_pos, :]
     all_atom_positions = all_atom_positions[..., ca_pos, :]
     all_atom_mask = all_atom_mask[..., ca_pos: (ca_pos + 1)]  # keep dim
-
-    chains = asym_id.unique()
-    one_hot = torch.nn.functional.one_hot(asym_id, num_classes=chains.shape[0]).to(dtype=all_atom_mask.dtype)
+    chains, _ = asym_id.unique(return_counts=True)
+    one_hot = torch.nn.functional.one_hot(asym_id.to(torch.int64)-1, # have to reduce asym_id by one because class values must be smaller than num_classes  
```

Comment:
Sorry I kept my modifications when I solved this conflict as asym_id starts from 1, we should deduct it by 1 so that the class values is always smaller than the number of classes. Otherwise, pytorch throws an error.
@christinaflo

Reply:
Oh yeah I was going to merge your changes in with this PR, I only wanted to remove the return_counts=True because I wasn't using the returned counts.